index.d.ts

```diff
@@ -1,5 +1,9 @@
 declare module 'react-string-replace' {
-  function reactStringReplace(text: React.ReactNode, regex: RegExp, cb: (match: string, index: number) => string | JSX.Element): JSX.Element;
+  function reactStringReplace(
+    text?: string | React.ReactNodeArray, 
+    regex?: string | RegExp, 
+    cb?: (match: string, index: number, offset: number) => React.ReactNode
+  ): React.ReactNodeArray;
```

Comment:
`reactStringReplace` always returns an array.
This array contains `string`s, which are one kind of `ReactNode` and anything thta might come from the callback, i.e. `ReactNode`.
Hence we always end up with an `ReactNodeArray`.